theseus/src/state/tags.rs

```diff
@@ -251,3 +243,56 @@ pub struct DonationPlatform {
     pub short: String,
     pub name: String,
 }
+
+#[derive(Debug, Clone, Decode, Encode, Serialize, Deserialize)]
+pub struct GameVersion {
+    pub version: String,
+    pub version_type: GameVersionString,
+    pub date: String,
+    pub major: bool,
+}
+
+// GameVersionString is a newtype wrapper around a string that allows it to be sorted by Minecraft version
+#[derive(Serialize, Deserialize, Decode, Encode, Clone, Debug, PartialEq)]
+#[serde(transparent)] // types that use this need to encode it as a String, so thats what we do
+pub struct GameVersionString(pub String);
```

Comment:
I doubt this will work as some game versions (like snapshots) don't have dots. I would just order by release date and use indexes.